packages/flutter_tools/lib/src/android/migrations/android_studio_java_gradle_conflict_migration.dart

```diff
@@ -0,0 +1,162 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../../base/file_system.dart';
+import '../../base/os.dart';
+import '../../base/platform.dart';
+import '../../base/process.dart';
+import '../../base/project_migrator.dart';
+import '../../base/version.dart';
+import '../../project.dart';
+import '../android_sdk.dart';
+import '../android_studio.dart';
+import '../gradle_utils.dart';
+
+// Android Studio 2022.2 "Flamingo" is the first to bundle a Java 17 JDK.
+// Previous versions bundled a Java 11 JDK.
+@visibleForTesting
+final Version androidStudioFlamingo = Version(2022, 2, 0);
+
+// Only match gradle.org urls so that we can guarantee that the modified url
+// actually points to a hosted zip.
+final RegExp _gradleVersionMatch =
```

Comment:
We had talked about making this a constant referenced from gradle_utils so that we could make sure that if one was update for something, like ignoring commented out lines, then the other would be as well.